vhost_user_backend/src/lib.rs

```diff
@@ -174,7 +174,11 @@ impl<S: VhostUserBackend> VhostUserDaemon<S> {
     /// terminate.
     pub fn wait(&mut self) -> Result<()> {
         if let Some(handle) = self.main_thread.take() {
-            handle.join().map_err(Error::WaitDaemon)?
+            match handle.join().map_err(Error::WaitDaemon)? {
```

Comment:
You need to remove `?`, otherwise you'll never enter the `Err()` case.

Reply:
Nope, the `.join()` returns the thing the thread returned which is itself a `Result` so we propagate the "join" error but interrogate the error that is actually returned by the thread.

Reply:
Ohhh I see! That's a tricky one :)